### ISSUE TITLE
Cleanup Structure

### ISSUE THREAD

**Issue by tracker1** [dropbox/zxcvbn#56]
Would suggest a preliminary cleanup of the project structure...

```
data/          - data directory
data-scripts/  - python scripts to load/parse data
dist/          - folder for target/compiled js
src/           - coffeescript & javascript source files
tests/         - unit tests (mocha?)
bower.json     - bower package info
package.json   - npm/node package information
```

`package.json` can reference `src/(index.js)` as the primary file.  `npm` scripts can be added to `package.json` for build, and/or use something like [gulp](http://gulpjs.com/).

Given that npm is already used for the installation of coffee-script and the nature of this package, it may be worthwhile to use [uglify-js](https://www.npmjs.org/package/uglify-js) over closure compiler.  It may not be quite as tight, but would reduce the need for external tooling (beyond node/npm) for this.

Converting the scripts to use CommonJS/node syntax could be used in combination with Browserify for the build, enabling a global and amd target in the dist directory from the same source.

I'd be happy to take this on, creating a fork and PR for the changes if there would be interest for including such changes/updates.

**Comment by avimar** [comment]
Maybe I'm just silly, but I couldn't figure out how to use this in NPM.
I went with a dependency on:
"zxcvbn": "git://github.com/buttercoin/node-zxcvbn.git#master"

... but I see there's few stars on that repo, or the other: https://github.com/mintplant/node-zxcvbn

Did I miss the simple way to use this server side with npm install?

**Comment by tracker1** [comment]
@avimar first, you will need to install it into your current environment, if you don't have a `package.json` run `npm init` to create one... then `npm install --save zxcvbn` will install and update your package.json... then in your code... `var zxcvbn = require('zxcvbn');` and use it per directions.

**Comment by avimar** [comment]
I did install that, which pulls from a different repo:
https://github.com/mintplant/node-zxcvbn/

But then doing var zxcvbn = require('zxcvbn') only gets me 
`Error: Cannot find module './zxcvbn/zxcvbn'`

It doesn't say in either the readme there or here how to use it inside node.js...


I went back to using: git://github.com/buttercoin/node-zxcvbn.git#master since that one works simply, the way you said this one should.

**Comment by stramel** [comment]
@tracker1 :+1:

**Comment by lowe** [comment]
Hi all, been a while but I figured better late than never! I now have more time to work on `zxcvbn` and have a fun roadmap planned.

@tracker1, I mostly stuck with your suggestions, check the v3 release. I used `lib` instead of `dist` for bundled and compiled .js following the lead of other node repos. 

I've also removed the java dependency, switching to uglifyjs instead of closure -- turns out gzipped difference is negligible even though it's over a 100k difference uncompressed. From what I can tell this is solely because closure replaces `["a", "b", "c"]` with a `"a,b,c".split` over zxcvbn's large dictionary lists -- an optimization we could do ourselves inside data-scripts/*.py.

Dan
